FEATURE: Add Zeal Wallet Package

resetko-zeal opened the pull request:
### Description
This PR separates the Zeal Wallet as a package that can be installed to ensure that it is always available as a wallet option. If the wallet is detected as already installed, onboarding will utilize the injected version. If it is not present and a user clicks the Zeal Wallet button, they will be redirected to the zeal website.

### Checklist
- [x] The version field in `package.json` is incremented following [semantic versioning](https://semver.org/)
- [X] The box that allows repo maintainers to update this PR is checked
- [X] I tested locally to make sure this feature/fix works
- [X] I have run `yarn file-check`, `yarn type-check` & `yarn build` to confirm there are not any associated errors
- [x] This PR passes the Circle CI checks

### Tests with demo app (injected)
- [X] sign message
- [x] sign typed message
- [x] send transaction
- [X] switch chains
- [X] disconnect

### Tests with demo app (SDK)
- [X] sign message
- [x] sign typed message
- [x] send transaction
- [X] switch chains
- [X] disconnect

resetko-zeal commented:
@Adamj1232 Hey, could you please help me finish this PR checklist :pray: 

Looking at [this PR](https://github.com/blocknative/web3-onboard/pull/1238/files) seems like I don't need to change versions, so probably should put a tick there?

Also a question regarding CI - probably I'd need some approval to run job to ensure that CI job is green (I'm actually don't have access to see what was the error message when I click "inspect")

Thanks in advance!

Adamj1232 commented on packages/zeal/package.json:
```diff
@@ -0,0 +1,59 @@
+{
+  "name": "@web3-onboard/zeal",
+  "version": "2.0.0",
```
```suggestion
  "version": "2.0.0-alpha.1",
```
This will publish a package to allow us to test prior to prod release

Adamj1232 commented on packages/demo/package.json:
```diff
@@ -45,6 +45,7 @@
     "@web3-onboard/enkrypt": "^2.0.0",
     "@web3-onboard/mew-wallet": "^2.0.0",
     "@web3-onboard/uauth": "^2.0.0",
+    "@web3-onboard/zeal": "^2.0.0",
```
```suggestion
    "@web3-onboard/zeal": "^2.0.0-alpha.1",
```

Adamj1232 commented on packages/zeal/README.md:
```diff
@@ -0,0 +1,30 @@
+
+### Install
+
+`npm i @web3-onboard/zeal`
```
```suggestion
`npm i @web3-onboard/core @web3-onboard/zeal`
```

Adamj1232 commented on packages/zeal/README.md:
```diff
@@ -0,0 +1,30 @@
+
+## Wallet module for connecting Zeal to web3-onboard
+
+See [Zeal](https://www.zeal.app/) for details
```
Please add more details here as this is usually first contact for developers

resetko-zeal replied:
:heavy_check_mark:

Adamj1232 commented:
@resetko-zeal Thanks for submitting this PR! 
I added some comments but generally looks good. Have you fully tested (**sign, send transaction, switch chains, disconnect without error**) Zeal with both injected and Zeal Module using the internal demo(deployed with `yarn dev`)?

resetko-zeal commented:
Hey @Adamj1232, I've added some more info to the `README.md`

Regarding the tests - I'd probably will need some help. So connect/disconnect, sign message works well, but other things, like signing typed data, sing transaction - I'm not able to make it work with any wallet (tried tally, rabby and zeal), despite the fact that in wild it works (with real dApps). Am I doing something wrong? I've only changed locally the `https: true` in the demo app config (because we do not allow our extension for regular `http` non-secure pages) and changed RPC URIs so they won't need keys (public ones). What data you usually use for testing in that case?

I've added also my test results to the PR header

Adamj1232 commented:
@resetko-zeal if you merge in the latest `v2-web3-onboard-develop` branch the send transaction should be fixed up.
I can look into `sign transaction` and `sign typed message`, we should be good to go as long as send is working properly and if the other two are working in the wild.
